generator/internal/descriptor_utils.cc

```diff
@@ -229,10 +235,12 @@ void SetMethodSignatureMethodVars(
       }
       method_signature += ", ";
     }
-    std::string key = "method_signature" + std::to_string(i);
+    if (field_deprecated) continue;
```

Comment:
Doesn't this mean that when an existing field is newly deprecated, we would generate a breaking change?